types/aws-lambda/index.d.ts

```diff
@@ -473,19 +473,18 @@ export interface Condition {
  * https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-policy-language-overview.html
  * https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html
  */
-export type Statement = BaseStatement & StatementAction & StatementResource;
+export type Statement = BaseStatement & StatementAction & (StatementResource | StatementPrincipal);
```

Comment:
This seems like it would permit invalid `{Not}Resource` values with valid `{Not}Principal` values, and vice versa - could you check with tests like:

```
// $ExpectError
statement = {
  Action: str,
  Principal: 123,
  Resource: str,
};
```